### PR TITLE
Add validator for discrete range of values

### PR DESCRIPTION
**Pull request by moritzj29**
Add validator to check if value is within a range of values and is a multiple of a given stepsize. This is useful for values which have a resolution limit. 
E.g. setting a sampling interval to 0.015s is allowed for a range of 0.01s to 2s. But if the maximum resolution of the instrument is 0.1s, it makes no sense to "pretend" higher accuracy in setting the value. This validator checks if the given value is a multiple of the resolution value or stepsize given.

### CODE REVIEW

File: pymeasure/instruments/validators.py [hunk 1]
```diff
@@ -40,6 +40,30 @@ def strict_range(value, values):
...
+    minimum of the range and is a multiple of step.
+    Otherwise it raises a ValueError.
+
+    :param value: A value to test
+    :param values: A range of values (range, list, etc.)
+    :param step: Minimum stepsize (resolution limit)
+    :raises: ValueError if the value is out of the range
+    """
+    if (min(values) <= value <= max(values)):
+        if (value/step).is_integer():
```
Comment (cjermain) [inline pymeasure/instruments/validators.py]: I think it is preferable to use the modulus to determine if it is a multiple: `value % step == 0`

File: pymeasure/instruments/validators.py [hunk 2]
```diff
@@ -40,6 +40,30 @@ def strict_range(value, values):
         ))
 
 
+def strict_discrete_range(value, values, step):
```
Comment (cjermain) [inline pymeasure/instruments/validators.py]: Please add tests in [test_validators.py](https://github.com/ralph-group/pymeasure/blob/master/tests/instruments/test_validators.py) to cover all use cases and edge cases.

**Comment by cjermain** [comment]
Please also add yourself to the Authors files:
https://github.com/ralph-group/pymeasure/blob/master/AUTHORS.txt
https://github.com/ralph-group/pymeasure/blob/master/docs/about/authors.rst

File: pymeasure/instruments/validators.py (hunk 1 above)
Comment (moritzj29) [reply pymeasure/instruments/validators.py]: This was my first intention as well, but I recognized that the modulus does not work very reliably with decimals:
`0.1 % 0.02 == 0` returns `False`
whereas the proposed
`(0.1/0.02).is_integer()` gives `True`
Do you have any idea how to fix this? It is probably due to some accuracy limit in number representation, since the modulus gives a very small (E-18) close to zero value.
The chosen variant is not as elegant as using the modulus but it works :D

Comment (bilderbuchi) [reply pymeasure/instruments/validators.py]: I thinks this is mainly down to floating point arithmetic. `math.isclose()` with appropriate tolerances is probably the safest/cleanest way to handle this.

Comment (moritzj29) [reply pymeasure/instruments/validators.py]: thanks for your comment! Upon your suggestion I tried `math.isclose()` and found a counterexample: `math.isclose(0,0.6%0.1,rel_tol=1e-9,abs_tol=1e-2)` evaluates to `False` despite the "huge" absolute tolerance. Are there any obejctions against the `Decimal` type solution? Computing time? As far as I understand this datatype is intended especially for such scenarios.

Comment (cjermain) [reply pymeasure/instruments/validators.py]: I think `Decimal` is a good solution for this use case. Thanks!
